nntrainer/layers/conv2d_layer.cpp

```diff
@@ -102,8 +102,8 @@ static void col2im(const Tensor &col_matrix, const TensorDim &kdim,
               continue;
             }
 
-            float *val = image.getAddress(0, c, h, w);
-            *val += col_matrix.getValue(0, 0, col_h, col_w);
+            float *val = (float *)image.getAddress(0, c, h, w);
```

Comment:
We should be using template T=float here instead of explicit typecast

Reply:
Currently wip for supporting various datatype cases for layers